### PR TITLE
Topic recognition - YOLOv1 on ISIC 2017/8 data set

### PR DESCRIPTION
**Pull request by Wolf-Syndrome**
Name - Lachlan Bunt
SID - s4698053
Task - Number 3, Detect lesions within the ISIC 2017/8 data set with YOLO

Files within folder recognition/YOLOv1 s4698053/:
.gitignore, README.md
images/ -> images for README.md
CONFIG.py -> Configuration for other files for easy access
dataset.py -> Used to load dataset from files to dataloader for torch to use
modules.py -> YOLOv1 model with some utility classes such as YOLO loss, save/load model
predict.py -> Used to create graphs and get evaluation of test dataset on trained model
train.py -> Used to train  YOLOv1 model in modules.py using dataset from dataset.py with CONFIG settings.

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme:  OK

- no training/testing loss or accuracy(mAP) plots shown
- minimal results and discussion

Commit messages:  OK

Code:

- No hyperparameter tuning or attempts to improve performance
- Minimal to no comments

Functionality/Performance:

- Average classification accuracy (precision?) is only 0.17
- Avergae IoU is only 0.39
- Poor performance

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, poor performance -1 classification not working (very low) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no training plots -2
Module present
Commenting minimal -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, missing refs and usage -2
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Need to remove gitignore file before I can merge. Does not affect grade only merge.

**Comment by Wolf-Syndrome** [comment]
@shakes76 Removed gitignore change

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.

**Comment by shakes76** [comment]
Sorry missed the pyc files which need to be removed still before merge. Only affects merge, grading is complete, no marks will be gained or lost.

**Comment by Wolf-Syndrome** [comment]
@shakes76 Removed pyc files
